src/MIDebugEngine/Engine.Impl/Variables.cs

```diff
@@ -343,23 +343,69 @@ public enum NodeType
 
         private string StripFormatSpecifier(string exp, out string formatSpecifier)
         {
-            formatSpecifier = null;
+            formatSpecifier = null; // will be used with -var-set-format
             int lastComma = exp.LastIndexOf(',');
-            if (lastComma > 0)
-            {
-                string expFS = exp.Substring(lastComma + 1);
-                string trimmed = expFS.Trim();
-                if (trimmed == "x" || trimmed == "X" || trimmed == "h" || trimmed == "H")
-                {
-                    formatSpecifier = "hexadecimal";
-                    return exp.Substring(0, lastComma);
-                }
-                else if (trimmed == "o")
-                {
+            if (lastComma <= 0)
+                return exp;
+
+            string expFS = exp.Substring(lastComma + 1);
+            string trimmed = expFS.Trim();
+            switch (trimmed)
+            {
+                case "x":
+                case "X":
```

Comment:
Yeah but is it worth it? In the end it's just displayed to the user.

Reply:
I am not sure the casing stuff really matters. I don't know why the `x` vs. `X` stuff was added. Maybe just because programmers are picky :).

I have seen folks really care about the no-quotes stuff through...

Reply:
Yeah to me it's only important that valid natvis doesn't get rejected.

Reply:
Yeah to me it's only important that valid natvis doesn't get rejected.

Reply:
I guess if you don't want to implement it, it would be acceptable to just add a comment indicating that this isn't supported yet, and what a real implementation should do.